src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java

```diff
@@ -65,91 +195,411 @@
-   * actionPlanId This method just creates a CaseNotification ======= Not sure this is the best
-   * place for this method, but .. several parts of case svc need to build a CaseNotification for a
-   * Case and need the services of the ActionPlanMappingService to get the actionPlanId This method
-   * just creates a CaseNotification >>>>>>> master
+   * Add new row to caseiacaudit table with casefk and iac of given case
    *
-   * @param caze The Case
-   * @param transitionEvent the event to inform the recipient of
-   * @return the newly created notification object
+   * @param updatedCase Case to create audit table row for
    */
-  CaseNotification prepareCaseNotification(Case caze, CaseDTO.CaseEvent transitionEvent);
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      readOnly = false,
+      timeout = TRANSACTION_TIMEOUT)
+  public void saveCaseIacAudit(final Case updatedCase) {
+    log.with("case_id", updatedCase.getId()).debug("Saving case iac audit");
+    CaseIacAudit caseIacAudit = new CaseIacAudit();
+    caseIacAudit.setCaseFK(updatedCase.getCasePK());
+    caseIacAudit.setIac(updatedCase.getIac());
+    caseIacAudit.setCreatedDateTime(DateTimeUtil.nowUTC());
+    caseIacAuditRepo.saveAndFlush(caseIacAudit);
+  }
 
   /**
-   * Create a CaseEvent from the details provided in the passed CaseEvent. Some events will also as
-   * a side effect create a new case - if so the details must be provided in the newCase argument,
-   * otherwise it may remain null. If the newCase is passed it must also contain the contact details
-   * for the new case.
+   * Overwrites existing iac for given case if iac is not active and adds new iac to caseiacaudit
+   * table
+   *
+   * @param targetCase Case to update
+   */
+  private void generateAndStoreNewIAC(final Case targetCase) {
+    String iac = targetCase.getIac();
+    if (iac == null || !internetAccessCodeSvcClient.isIacActive(iac)) {
+      log.with("case_id", targetCase.getId()).debug("Replacing existing case IAC");
+      String newIac = internetAccessCodeSvcClient.generateIACs(1).get(0);
+      targetCase.setIac(newIac);
+      caseRepo.saveAndFlush(targetCase);
+      saveCaseIacAudit(targetCase);
+    } else {
+      log.with("case_id", targetCase.getId()).debug("Existing IAC is still active");
+    }
+  }
+
+  private void transitionCaseGroupStatus(final Case targetCase, final CaseEvent caseEvent) {
+    CaseGroup caseGroup = caseGroupRepo.findOne(targetCase.getCaseGroupFK());
+    try {
+      caseGroupService.transitionCaseGroupStatus(
+          caseGroup, caseEvent.getCategory(), targetCase.getPartyId());
+    } catch (CTPException e) {
+      // The transition manager throws an exception if the event doesn't cause a transition, however
+      // there are lots of
+      // events which do not cause CaseGroupStatus transitions, (this is valid behaviour).
+      log.debug(e.getMessage());
```

Comment:
I am deeply dubious about this catch without a rethrow and no error logged. Seems like a route to flakey/unpredictable behaviour